### ISSUE TITLE
Different object fill patterns

### ISSUE THREAD

**Issue by Arcanorum** [bjorn/tiled#1681]
It would be cool if we could pick different patterns for the fill of objects.

For example, if I have a lot of object types, I quickly run out of different colours that are distinct enough from each other. If there were patterns (such as dots, a grid, lines in various directions, etc.) that could be added to the fill, they could be differentiated like that.

![patterns](https://user-images.githubusercontent.com/9785268/29009206-085a61d2-7b19-11e7-95d5-b0913fb6c9b3.PNG)

**Comment by akomarovskih** [comment]
I agree with Arcanorum - it would be cool to have patterns or feature to set image icon for every type of objects. Сolored frame is not enough to visually identify the type of object on map.

**Comment by bjorn** [comment]
I like the suggestion as well, but if you need an image icon and don't care about the size of the object, then you could use tile objects instead.
